lib/core/utils/filter-html-attrs.js

```diff
@@ -0,0 +1,178 @@
+// grab everything between an opening and closing bracket
+const tagStringRegex = /<([^>]+?)\/?>/g;
+
+/*
+  match attributes. html does not allow escaping quotes inside attributes so we don't have to worry about that (e.g. aria-label="\"label\"").
+  @see https://regexr.com/5mbqb
+*/
+const attributeRegex = new RegExp(
+  `(?:([^\\s=]+)=` + // group 1: valued attribute name; look for non-whitespace, non-equal sign characters followed by the "=" sign
+  `(?:"([^"]*)"` + // group 2: double quoted attribute value; look for everything surrounded by double quotes
+  `|'([^']*)'` + // group 3: single quoted attribute value; look for everything surrounded by single quotes
+  `|([^'"\\s]*)))` + // group 4: non-quoted attribute value; look for any non-whitespace, non-quote characters
+    `|\\b(\\S+)(?:\\B|\\b)`, // group 5: non-valued attribute name; look for any remaining non-matched strings surrounded by word boundaries
+  'g'
+);
+
+const attrMap = {
+  for: 'htmlFor',
+  class: 'className'
+};
+
+/**
+ * Match an attribute to a matcher object.
+ * @param {String} nodeName - Name of the current node
+ * @param {Object[]} attributes - Array of node attributes and values
+ * @param {String} value - Value of the current attribute
+ * @param {any} matcher - How to match the value
+ */
+function matches(nodeName, attributes, value, matcher) {
+  switch (typeof matcher) {
+    case 'boolean':
+      return true;
+    case 'string':
+      return matcher === value;
+    case 'object':
+      if (Array.isArray(matcher)) {
+        return matcher.includes(value);
+      }
+
+      // matcher-like object, match all parts
+      let filter = true;
+
+      // match each property
+      if (matcher.nodeName) {
+        filter = filter && matcher.nodeName === nodeName;
```

Comment:
This looks a little more readable:

```suggestion
      if (matcher.nodeName && matcher.nodeName !== nodeName) {
        return false
```